java-generator/core/src/main/java/io/fabric8/java/generator/nodes/JObject.java

```diff
@@ -208,6 +208,16 @@ public GeneratorResult generateJava() {
                   new StringLiteralExpr(
                       prop.getDescription().replace("\"", "\\\""))));
         }
+
+        if (!prop.isNullable) {
+          // from https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#defaulting-and-nullable :
+          // "_null values for fields that either don't specify the nullable flag, or give it a false
+          // value, will be pruned before defaulting happens. If a default is present, it will be applied._"
+          objField.addAnnotation(
+              new SingleMemberAnnotationExpr(
+                  new Name("com.fasterxml.jackson.annotation.JsonSetter"),
+                  new NameExpr("nulls=com.fasterxml.jackson.annotation.Nulls.SKIP")));
```

Comment:
this encoding looks correct to me, but I would like someone else opinion to be sure, cc. @manusa @metacosm

Reply:
This means that the default value is preserved in case it existed which should be aligned with what's stated in this epigraph: https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#defaulting-and-nullable